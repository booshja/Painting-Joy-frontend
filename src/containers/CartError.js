// dependencies
import React from "react";
import styled from "styled-components";
// components
import { GoBack, PageTitle } from "../components";
// assets
import OopsImg from "../assets/jelleke-vanooteghem-kabtmcdcAbk-unsplash.jpg";
// breakpoints
import { breakpoints } from "../breakpoints";

const StyledCartError = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
`;

const StyledP = styled.div`
    font-family: "News Cycle", sans-serif;
    font-size: 1.2rem;
    text-align: center;
    line-height: 1.2;
    width: 80%;
    margin-bottom: 1rem;

    ${breakpoints("font-size", "rem", [{ 414: 1.4 }])}
`;

const StyledImg = styled.img`
    width: 80%;
`;

const CartError = () => {
    // error page for items out of stock in cart upon checkout
    return (
        <StyledCartError>
            <PageTitle>Uh Oh</PageTitle>
            <StyledP>
                Looks like someone else snagged something you were looking at
                first, sorry!
            </StyledP>
            <StyledP>
                We've updated your cart, go ahead and check it out.
            </StyledP>
            <GoBack to="/cart" />
            <StyledImg
                src={OopsImg}
                alt="Child with hands over their mouth with an oops face"
            />
        </StyledCartError>
    );
};

export default CartError;
