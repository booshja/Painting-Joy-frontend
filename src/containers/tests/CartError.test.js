import { render } from "@testing-library/react";
import CartError from "../CartError";

/** Smoke Test */
it("renders without crashing", () => {
    render(<CartError />);
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(<CartError />);
    expect(asFragment()).toMatchSnapshot();
});
