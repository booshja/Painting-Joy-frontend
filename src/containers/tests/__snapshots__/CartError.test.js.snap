// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-eWnToP fcnWwj"
  >
    <h1
      class="sc-fWWYYk cwzJaj"
    >
      Uh Oh
    </h1>
    <div
      class="sc-kTCsyW cTCUUg"
    >
      Looks like someone else snagged something you were looking at first, sorry!
    </div>
    <div
      class="sc-kTCsyW cTCUUg"
    >
      We've updated your cart, go ahead and check it out.
    </div>
    <button
      class="sc-fXgAZx fGPCkt"
    >
      <i
        class="fas fa-long-arrow-alt-left"
      />
      Back
    </button>
    <img
      alt="Child with hands over their mouth with an oops face"
      class="sc-dvUynV jVrjIS"
      src="jelleke-vanooteghem-kabtmcdcAbk-unsplash.jpg"
    />
  </div>
</DocumentFragment>
`;
